src/main/java/org/springframework/data/tarantool/core/DefaultTarantoolExceptionTranslator.java

```diff
@@ -26,8 +23,8 @@ public DataAccessException translateExceptionIfPossible(RuntimeException cause)
             if (cause instanceof TarantoolClientException) {
                 return new InvalidDataAccessResourceUsageException(cause.getMessage(), cause);
             }
-            if (cause instanceof TarantoolServerException || cause instanceof TarantoolFunctionCallException) {
-                return new DataRetrievalFailureException(cause.getMessage(),  cause);
+            if (cause instanceof TarantoolInternalException || cause instanceof TarantoolFunctionCallException) {
```

Comment:
TarantoolFunctionCallException now is more related to the upper case (InvalidDataAccessResourceUsageException)